From: Reinhard Tartler <siretart@tauware.de>
Date: Sat, 13 Jul 2024 08:22:13 -0400
Subject: Relax dependencies

---
 Cargo.toml | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/Cargo.toml b/Cargo.toml
index 1071079..f6a11bf 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -23,8 +23,8 @@ platforms = ["x86_64-unknown-linux-gnu", "aarch64-unknown-linux-gnu", "powerpc64
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-clap = { version = "~4.5.40", features = ["derive"] }
-syslog = "^7.0.0"
+clap = { version = "~4.5.23", features = ["derive"] }
+syslog = "6.0.0"
 log = "0.4.27"
 hickory-server = "0.25.2"
 hickory-proto = { version = "0.25.2", features = ["tokio"] }
@@ -37,4 +37,4 @@ arc-swap = "1.7.1"
 flume = "0.11.1"
 
 [build-dependencies]
-chrono = "0.4.41"
+chrono = "0.4.39"
