From: Reinhard Tartler <siretart@tauware.de>
Date: Sat, 13 Jul 2024 08:22:13 -0400
Subject: Relax dependencies

---
 Cargo.toml | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

Index: aardvark-dns/Cargo.toml
===================================================================
--- aardvark-dns.orig/Cargo.toml
+++ aardvark-dns/Cargo.toml
@@ -22,8 +22,8 @@ platforms = ["x86_64-unknown-linux-gnu",
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-clap = { version = "~4.4.10", features = ["derive"] }
-syslog = "^7.0.0"
+clap = { version = ">= 4.3", features = ["derive"] }
+syslog = ">= 6.0"
 log = "0.4.22"
 hickory-server = "0.24.1"
 hickory-proto = { version = "0.24.1", features = ["tokio-runtime"] }
