From: Reinhard Tartler <siretart@tauware.de>
Description: Make it compile with trust-dns 0.21

diff --git a/src/dns/coredns.rs b/src/dns/coredns.rs
index 0eb922a..e83e0a6 100644
--- a/src/dns/coredns.rs
+++ b/src/dns/coredns.rs
@@ -1,8 +1,10 @@
 use crate::backend::DNSBackend;
 use crate::backend::DNSResult;
 use futures_util::StreamExt;
+use futures_util::TryStreamExt;
 use log::{debug, error, trace, warn};
 use resolv_conf;
+use std::convert::TryInto;
 use std::env;
 use std::fs::File;
 use std::io::Read;
@@ -14,8 +16,8 @@ use trust_dns_proto::{
     op::{Message, MessageType, ResponseCode},
     rr::{DNSClass, RData, Record, RecordType},
     udp::{UdpClientStream, UdpStream},
-    xfer::{dns_handle::DnsHandle, DnsRequest},
-    BufStreamHandle,
+    xfer::{dns_handle::DnsHandle, DnsRequest, BufDnsStreamHandle},
+    DnsStreamHandle,
 };
 
 pub struct CoreDns {
@@ -103,7 +105,8 @@ impl CoreDns {
 
         // Do we need to serve on tcp anywhere in future ?
         let socket = UdpSocket::bind(format!("{}:{}", self.address, self.port)).await?;
-        let (mut receiver, sender) = UdpStream::with_bound(socket);
+        let address = SocketAddr::new(self.address, self.port.try_into().unwrap());
+        let (mut receiver, sender) = UdpStream::with_bound(socket, address);
 
         loop {
             tokio::select! {
@@ -336,7 +339,7 @@ impl CoreDns {
     }
 }
 
-fn reply(mut sender: BufStreamHandle, socket_addr: SocketAddr, msg: &Message) -> Option<()> {
+fn reply(mut sender: BufDnsStreamHandle, socket_addr: SocketAddr, msg: &Message) -> Option<()> {
     let id = msg.id();
     let mut msg_mut = msg.clone();
     msg_mut.set_message_type(MessageType::Response);
@@ -390,8 +393,8 @@ async fn forward_dns_req(mut cl: AsyncClient, message: Message) -> Option<Messag
     let req = DnsRequest::new(message, Default::default());
     let id = req.id();
 
-    match cl.send(req).await {
-        Ok(mut response) => {
+    match cl.send(req).try_next().await {
+        Ok(Some(mut response)) => {
             response.set_id(id);
             for answer in response.answers() {
                 debug!(
@@ -405,6 +408,10 @@ async fn forward_dns_req(mut cl: AsyncClient, message: Message) -> Option<Messag
             }
             Some(response.into())
         }
+        Ok(None) => {
+            error!("{} dns request got empty response", id);
+            None
+        }
         Err(e) => {
             error!("{} dns request failed: {}", id, e);
             None
